src/diagnostics/diagnostic.jl

```diff
@@ -661,6 +661,37 @@ function accumulate!(diag_accumulator, diag_storage, reduction_time_func)
     return nothing
 end
 
+any_sync_needed(step, diagnostics) =
+    any(diag -> sync_needed(step, diag), diagnostics)
+
+# sync_needed(step, diag) = diag.cbf.n > 0 && step % diag.cbf.n == 0
+sync_needed(step, diag) = diag.output_every > 0 && step % diag.output_every == 0
+
+import NVTX
+NVTX.@annotate function sync_nc_datasets(step, diagnostics, output_dir)
+    any(diag -> diag.output_writer isa NetCDFWriter, diagnostics) || return
+    any_sync_needed(step, diagnostics) || return
+    tasks = map(diagnostics) do diag
+        writer = diag.output_writer
+        if writer isa NetCDFWriter && sync_needed(step, diag)
+            output_path = outpath_name(output_dir, diag)
+            # TODO: find out why Threads.@spawn segfaults in multithreaded IO test
```

Comment:
Something we might want to keep in mind is this:

https://juliaparallel.org/MPI.jl/stable/reference/environment/#MPI.ThreadLevel

Reply:
(It should not be a problem at the moment because sync does not contain an MPI call.)

Reply:
Yeah, I thought that only the root process does this, but it's a good point.